### PR TITLE
[SOAR-1934] New Action Insight IDR Get a Saved Query

### PR DESCRIPTION
**Pull request by ablakley-r7**
### Description

Describe the proposed changes:

  - Add new action Get a Saved Query which returns a saved LEQL query and accepts the query ID as a parameter

## PR Requirements

Developers, verify you have completed the following items by checking them off:

#### Unit Tests

Review our documentation on [generating](https://docs.rapid7.com/insightconnect/unit-test-generation) and [writing](https://docs.rapid7.com/insightconnect/unit-test-primer) plugin unit tests

- [x] Unit tests written for any new or updated code

#### In-Product Tests

If you are an InsightConnect customer or have access to an InsightConnect instance, the following in-product tests should be done:

- [x] Screenshot of job output with the plugin changes
- [x] Screenshot of the changed connection, actions, or triggers input within the InsightConnect workflow builder

Screenshot 2022-05-05 at 11 32 23

Screenshot 2022-05-05 at 12 27 33

### Style

Review the [style guide](https://docs.rapid7.com/insightconnect/style-guide/)

- [x] For dependencies, pin [OS package](https://docs.rapid7.com/insightconnect/style-guide/#dockerfile) and [Python package](https://docs.rapid7.com/insightconnect/style-guide/#requirements.txt) versions
- [x] For security, set least privileged account with ``USER nobody`` in the ``Dockerfile`` when possible
- [x] For size, use the [slim SDK images](https://docs.rapid7.com/insightconnect/sdk-guide/#sdk-guide) when possible: ``komand/python-3-37-slim-plugin`` and ``komand/python-3-37-plugin``
- [x] For error handling, use of [PluginException](https://docs.rapid7.com/insightconnect/error-handling-in-integrations/#plugin-exceptions) and [ConnectionTestException](https://docs.rapid7.com/insightconnect/error-handling-in-integrations#connection-exceptions)
- [x] For logging, use [self.logger](https://docs.rapid7.com/insightconnect/sdk-guide/#logging)
- [x] For docs, use [changelog style](https://docs.rapid7.com/insightconnect/style-guide/#changelog)
- [x] For docs, validate markdown with ``make validate`` which calls ``mdl`` to lint ``help.md``

### Functional Checklist
- [x] Work fully completed
- [x] Functional
  - [x] Any new actions/triggers include JSON [test files](https://docs.rapid7.com/insightconnect/style-guide/#tests) in the `tests/` directory created with `icon-plugin run -c sample $action > tests/$action.json`
  - [ ] Tests should all pass unless it's a negative test. Negative tests have a naming convention of `tests/$action_bad.json`
  - [ ] Unsuccessful tests should fail by raising an exception causing the plugin to die and an object should be returned on successful test
  - [ ] Add functioning test results to PR, sanitize any output if necessary
    * Single action/trigger `icon-plugin run -T tests/example.json --debug --jq`
    * All actions/triggers shortcut `icon-plugin run -T all --debug --jq` (use PR format at end)
  - [ ] Add functioning run results to PR, sanitize any output if necessary
    * Single action/trigger `icon-plugin run -R tests/example.json --debug --jq`
    * All actions/triggers shortcut `icon-plugin run -R all --debug --jq` (use PR format at end)

### Assessment

You must validate your work to reviewers:

1. Run `make validate` and make sure everything passes
2. Run the assessment tool: `icon-plugin run -A -R all -T all`. For single action validation: `icon-plugin run -A -R tests/my_action.json -T tests/my_action.json`
3. Copy (`icon-plugin ... | pbcopy`) and paste the output in **a new post** on this PR
4. Add required screenshots from the In-Product Tests section

### CODE REVIEW

File: plugins/rapid7_insightidr/help.md [hunk 1]
```diff
@@ -40,6 +40,66 @@ Example input:
...
+
+##### Output
+
+|Name|Type| Required |Description|
+|----|----|----|-----------|
+|saved_query|query|True|Saved LEQL query|
+
+```
+{
+  "saved_query": {
```
Comment (igorski-r7) [inline plugins/rapid7_insightidr/help.md:83]: Doubled output here 😃

**Review by igorski-r7 — APPROVED** [review]

File: plugins/rapid7_insightidr/help.md (hunk 1 above)
Comment (ablakley-r7) [reply plugins/rapid7_insightidr/help.md]: Fixed now!

File: plugins/rapid7_insightidr/help.md [hunk 2]
```diff
@@ -25,7 +25,7 @@ The connection configuration accepts the following parameters:
 |Name|Type|Default|Required|Description|Enum|Example|
 |----|----|-------|--------|-----------|----|-------|
 |api_key|credential_secret_key|None|True|InsightIDR API key|None|4472f2g7-991z-4w70-li11-7552w8qm0266|
-|url|string|https://us.api.insight.rapid7.com|True|The URL endpoint for InsightIDR. e.g. https://<REGION_CODE>.api.insight.rapid7.com|None|https://us.api.insight.rapid7.com|
+|url|string|https://example.com|True|The URL endpoint for InsightIDR. e.g. https://<REGION_CODE>.api.insight.rapid7.com|None|https://example.com|
```
Comment (cmcnally-r7) [inline plugins/rapid7_insightidr/help.md:28]: Move the example to the example field

File: plugins/rapid7_insightidr/plugin.spec.yaml [hunk 3]
```diff
@@ -350,6 +351,54 @@ types:
...
+      description: Time range start value
+      required: false
+  leql:
+    statement:
+      title: Statement
+      type: string
+      description: Logentries Query Language statement
+      required: false
+    during:
+      title: Statement
```
[the remark this answers is not in the document]
Comment (ablakley-r7) [reply plugins/rapid7_insightidr/plugin.spec.yaml]: Definitely, this is now updated.

File: plugins/rapid7_insightidr/plugin.spec.yaml [hunk 4]
```diff
@@ -350,6 +351,54 @@ types:
...
+    from:
+      title: From
+      type: integer
+      description: Time range start value
+      required: false
+  leql:
+    statement:
+      title: Statement
+      type: string
+      description: Logentries Query Language statement
```
[the remark this answers is not in the document]
Comment (ablakley-r7) [reply plugins/rapid7_insightidr/plugin.spec.yaml:374]: I think this should remain capitalised as it looks like a proper noun.
[Log Entries](https://docs.logentries.com/docs/search)

File: plugins/rapid7_insightidr/help.md [hunk 5]
```diff
@@ -805,10 +799,11 @@ _This plugin does not contain any troubleshooting information._
 
 # Version History
 
+* 3.2.0 - Add new action Get A Saved Query
 * 3.1.4 - Add `docs_url` to plugin spec with a link to [InsightIDR plugin setup guide](https://docs.rapid7.com/insightconnect/rapid7-insightidr)
 * 3.1.3 - Fix issue where Get a Log and Get All Logs would either fail in workflow or in connection test
 * 3.1.2 - Send plugin name and version in the User-Agent string to vendor
-* 3.1.1 - Convert given date from timezone to UTC in List Investigations action
+* 3.1.1 - Convert given date from timezone to UTC in List Investigataions action
```
[the remark this answers is not in the document]
Comment (ablakley-r7) [reply plugins/rapid7_insightidr/help.md]: Updated.

File: plugins/rapid7_insightidr/help.md (hunk 2 above)
Comment (ablakley-r7) [reply plugins/rapid7_insightidr/help.md]: Have reverted this which was changed unintentionally by a regeneration. I think the example in the description field might be alright as it gives instruction on the region code used in the url.

**Review by cmcnally-r7 — APPROVED** [review]

**Pull request merged**